### DISCUSSION TITLE
GLIBC_2.29 not found when loading STT under docker image

### DISCUSSION THREAD

**Discussion by alpoktem** [coqui-ai/STT#1884] [Q&A] [answered]
Hi, 

I had STT python module successfully installed on my machine. However, I am having trouble with installing and loading under my docker image. This is my Dockerfile:

```
FROM python:3.8-slim

WORKDIR /app

COPY ./requirements.txt /app/requirements.txt

RUN pip install -r /app/requirements.txt \
    && rm -rf /root/.cache/pip

COPY . /app/
```

STT is specified in `requirements.txt` as `stt==0.10.0a6`.

My docker container builds but when I run I get this:
```
asr_1  |   File "/app/./app/api/transcribeAPI.py", line 239, in load_models
asr_1  |     import stt
asr_1  |   File "/usr/local/lib/python3.8/site-packages/stt/__init__.py", line 23, in <module>
asr_1  |     from stt.impl import Version as version
asr_1  |   File "/usr/local/lib/python3.8/site-packages/stt/impl.py", line 13, in <module>
asr_1  |     from . import _impl
asr_1  | ImportError: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.29' not found (required by /usr/local/lib/python3.8/site-packages/stt/lib/libstt.so)
```

I appreciate any guidance.

**Comment by reuben** [comment]
Sounds like our build environment is not compatible with the `python:3.8-slim` target image. A possible easy fix you can try is changing the `build-python-Linux` CI task [0] on your fork to use an older runner, eg Ubuntu 18.04 or 16.04 [1] and seeing if that helps.

The correct fix is to use the official manylinux Docker images [2] in order to create the Python packages.

[0] https://github.com/coqui-ai/STT/blob/4ebcbea8b3a78d8df79eabc384799d635f9ba8f2/.github/workflows/build-and-test.yml#L310
[1] https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners
[2] https://github.com/pypa/manylinux

**Comment by alpoktem** [comment, marked as answer]
Thanks @reuben. Using Ubuntu image did the job. Posting here the working Dockerfile

```
FROM ubuntu:20.04

WORKDIR /app

COPY ./requirements.txt /app/requirements.txt

RUN apt-get update && apt-get clean

RUN apt-get install -y python3-pip

RUN pip install -r /app/requirements.txt \
    && rm -rf /root/.cache/pip

COPY . /app/
```
